### PR TITLE
Fix metadata, call caching issues for CWL Long type

### LINKED ISSUE #4023: CWL restart problem when reading from cache -- unrecognized simpleton WOM type: Long
Hi;
I'm running Cromwell 34 and hitting an issue when reading from a cache. The problematic CWL step uses a variable, `min_allele_fraction` that is initially an array of longs:

https://github.com/bcbio/test_bcbio_cwl/blob/eae685b8023126b7f159d3048f4ab4dd1a1833d6/prealign/prealign-workflow/steps/batch_for_variantcall.cwl#L101

and then gets converted into a record with individual long elements:

https://github.com/bcbio/test_bcbio_cwl/blob/eae685b8023126b7f159d3048f4ab4dd1a1833d6/prealign/prealign-workflow/steps/batch_for_variantcall.cwl#L309

This all works fine on the first run of a pipeline, but when reading the step from the cache we get an error about not supporting `Long`:
```
[2018-08-21 10:26:40,02] [info] WorkflowExecutionActor-3e76006c-870a-4c34-9f21-949eee1a5b33 [3e76006c]: Starting batch_for_variantcall
[2018-08-21 10:26:41,10] [error] unrecognized simpleton WOM type: Long
java.lang.RuntimeException: unrecognized simpleton WOM type: Long
        at cromwell.Simpletons$.toSimpleton(Simpletons.scala:30)
        at cromwell.Simpletons$.toSimpleton(Simpletons.scala:16)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.$anonfun$new$2(FetchCachedResultsActor.scala:32)
        at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
        at scala.collection.Iterator.foreach(Iterator.scala:944)
        at scala.collection.Iterator.foreach$(Iterator.scala:944)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at scala.collection.TraversableLike.map(TraversableLike.scala:234)
        at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
        at scala.collection.AbstractTraversable.map(Traversable.scala:104)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.$anonfun$new$1(FetchCachedResultsActor.scala:32)
        at cromwell.engine.workflow.lifecycle.execution.callcaching.FetchCachedResultsActor.$anonfun$new$1$adapted(FetchCachedResultsActor.scala:30)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:60)
        at akka.dispatch.BatchingExecutor$AbstractBatch.processBatch(BatchingExecutor.scala:55)
        at akka.dispatch.BatchingExecutor$BlockableBatch.$anonfun$run$1(BatchingExecutor.scala:91)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
        at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:81)
        at akka.dispatch.BatchingExecutor$BlockableBatch.run(BatchingExecutor.scala:91)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:44)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```
I can reproduce with this test CWL workflow (https://github.com/bcbio/test_bcbio_cwl/tree/master/prealign). The first time running will work fine, and then if you re-run using the cache from the previous run it fails as above.

Thanks for any thoughts/tips about what is going wrong here and how best to fix.

### PR DESCRIPTION
**Pull request by aednichols**
Closes #4023 

I hijacked one of the steps in `cwl_cache_between_workflows.cwl` to test `Long` alongside `Float` - the second commit [fails](https://api.travis-ci.org/v3/job/419242749/log.txt) with the error the user saw.

### CODE REVIEW

File: centaur/src/main/resources/standardTestCases/cwl_cache_between_workflows/cwl_cache_between_workflows.cwl [hunk 1]
```diff
@@ -36,7 +36,7 @@ $graph:
 
   outputs:
     trapezoidalArea:
-      type: float
+      type: long
```
Comment (Horneth) [inline centaur/src/main/resources/standardTestCases/cwl_cache_between_workflows/cwl_cache_between_workflows.cwl]: Are we still testing `float` somewhere else ?

Comment (aednichols) [reply centaur/src/main/resources/standardTestCases/cwl_cache_between_workflows/cwl_cache_between_workflows.cwl]: Should be - `baseAverage` still uses `float` and and its value appears in the final outputs (would love a cross-check on this thinking from a CWL expert)

```
{
  "cwl-cache-between-workflows.trapezoidalArea": "77",
  "cwl-cache-between-workflows.baseAverage": 11.0
}
```

Comment (aednichols) [reply centaur/src/main/resources/standardTestCases/cwl_cache_between_workflows/cwl_cache_between_workflows.cwl]: Checking this now on branch [`aen_4023_testing_the_test`](https://github.com/broadinstitute/cromwell/compare/aen_4023_testing_the_test?expand=1) where I deliberately break `float`

Comment (aednichols) [reply centaur/src/main/resources/standardTestCases/cwl_cache_between_workflows/cwl_cache_between_workflows.cwl]: Yep, we get `java.lang.RuntimeException: unrecognized simpleton WOM type: Float` as expected

**Pull request merged**